src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,571 @@
+      it("should return Infinity in getMaxBufferBehind by default", () => {
+        expect(player.getMaxBufferBehind()).to.equal(Infinity);
+      });
+    });
+
+    describe("getMaxBufferAhead", () => {
+      it("should return Infinity in getMaxBufferAhead by default", () => {
+        expect(player.getMaxBufferAhead()).to.equal(Infinity);
+      });
+    });
+
+    describe("getPlaybackRate/setPlaybackRate", () => {
+      it("should allow to change the playback rate through setPlaybackRate", () => {
+        expect(player.setPlaybackRate(4)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(4);
+
+        expect(player.setPlaybackRate(3)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(3);
+
+        expect(player.setPlaybackRate(2)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(2);
+
+        expect(player.setPlaybackRate(1.5)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(1.5);
+
+        expect(player.setPlaybackRate(0.7)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(0.7);
+
+        expect(player.setPlaybackRate(1)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(1);
+      });
+    });
+
+    describe("seekTo", () => {
+      it("should throw in seekTo by default", () => {
+        expect(() => player.seekTo(10)).to.throw();
+        expect(() => player.seekTo(54)).to.throw();
+        expect(() => player.seekTo({ relative: 5 })).to.throw();
+        expect(() => player.seekTo({ position: 5 })).to.throw();
+        expect(() => player.seekTo({ wallClockTime: 5 })).to.throw();
+      });
+    });
+
+    describe("exitFullscreen", () => {
+      it("should allow exitFullscreen by default", () => {
+        /* tslint:disable deprecation */
+        expect(player.exitFullscreen()).to.equal(undefined);
+        /* tslint:enable deprecation */
+      });
+    });
+
+    describe("setFullscreen", () => {
+      it("should allow setFullscreen by default", () => {
+        /* tslint:disable deprecation */
+        expect(player.setFullscreen()).to.equal(undefined);
+        /* tslint:enable deprecation */
+
+        // TODO remove for v3.0.0
+        /* tslint:disable deprecation */
+        expect(player.setFullscreen(false)).to.equal(undefined);
+        /* tslint:enable deprecation */
+      });
+    });
+
+    describe("getVolume/setVolume", () => {
+      it("should throw in setVolume by default if no volume has been given", () => {
+        expect(() => player.setVolume(5)).to.throw();
+      });
+
+      /* tslint:disable:max-line-length */
+      it("should set the volume in setVolume by default if a volume has been given", () => {
+      /* tslint:enable:max-line-length */
+        const videoElement = player.getVideoElement();
+        if (videoElement == null) {
+          throw new Error("The API is disposed");
+        }
+        expect(player.setVolume(1)).to.equal(undefined);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this use of the output from "setVolume"; "setVolume" doesn't return anything. ([typescript:S3699](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3699))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=437&issues=AWgOIuUCUMkE2J58175u&open=AWgOIuUCUMkE2J58175u)

[ISSUE_KEY]: AWgOIuUCUMkE2J58175u